proto/cellar.proto

```diff
@@ -0,0 +1,230 @@
+/*
+* An API for cellars that implement the EIP4626-compatible Cellar.sol contract.
+*
+* See [Cellar.sol](https://github.com/PeggyJV/cellar-contracts/blob/main/src/base/Cellar.sol)
+*/
+
+syntax = "proto3";
+package steward.v2;
+
+option go_package = "/steward_proto";
+
+/*
+* Represents a function call to a cellar that implements Cellar.sol
+*/
+message Cellar {
+    // The function you wish to execute on the target cellar
+    oneof function {
+        // Represents function `addPosition(uint256 index, address position)`
+        AddPosition add_position = 1;
+        // Represents function `pushPosition(address position)`
+        PushPosition push_position = 2;
+        // Represents function `removePosition(uint256 index)`
+        RemovePosition remove_position = 3;
+        // Represents function `setHoldingPosition(address newHoldingPosition)`
+        SetHoldingPosition set_holding_position = 4;
+        /*
+        * Represents function `rebalance(address fromPosition, address toPosition,
+            uint256 assetsFrom, SwapRouter.Exchange exchange, bytes calldata params)`
+        */
+        Rebalance rebalance = 5;
+        // Represents function `setStrategistPayoutAddress(address payout)`
+        SetStrategistPayoutAddress set_strategist_payout_address = 6;
+        // Represents function `setWithdrawType(WithdrawType newWithdrawType)`
+        SetWithdrawType set_withdraw_type = 7;
+        // Represents function `swapPositions(uint256 index1, uint256 index2)`
+        SwapPositions swap_positions = 8;
+        // Represents function `setDepositLimit()`
+        SetDepositLimit set_deposit_limit = 9;
+        // Represents function `setLiquidityLimit()`
+        SetLiquidityLimit set_liquidity_limit = 10;
+        // Represents function `setShareLockPeriod()`
+        SetShareLockPeriod set_share_lock_period = 11;
+        // Represents function `setRebalanceDeviation(uint265)`
+        SetRebalanceDeviation set_rebalance_deviation = 12;
+    }
+
+    /*
+    * Insert a trusted position to the list of positions used by the cellar at a given index.
+    *
+    * Represents function `addPosition(uint256 index, address position)`
+    */
+    message AddPosition {
+        // Index at which to add the position
+        uint64 index = 1;
```

Comment:
Are `uint64` types throughout our definitions here sufficient to represent what will be stored in the `uint256` values in the contract on a case-by-case basis?

Reply:
I can see it going either way. Should we just always default to string if it's uint256?

Reply:
The Aave cellar has a mix, but this is an opportunity to change that. It really depends on the expected values for the argument -- leaving it as a string puts no restrictions on it but takes up more wire space, though relatively negligible in this context. If the contract team can confirm the value will always exist within the bounds of the max size of the input (e.g. for `uint64`) then it should be OK, but in truth I think it might be safer to just universally handle all `uint256` the same.

Reply:
Let's go with the latter